frontend/pkg/frontend/ocm.go

```diff
@@ -140,13 +140,13 @@ func (f *Frontend) BuildCSCluster(resourceID *arm.ResourceID, tenantID string, h
 		"np_provisioner_provision_enabled":   "true",
 		"np_provisioner_deprovision_enabled": "true",
 	}
-	if f.clusterServiceConfig.ProvisionShardID != nil {
-		additionalProperties["provision_shard_id"] = *f.clusterServiceConfig.ProvisionShardID
+	if f.clusterServiceClient.GetProvisionShardID() != nil {
+		additionalProperties["provision_shard_id"] = *f.clusterServiceClient.GetProvisionShardID()
 	}
-	if f.clusterServiceConfig.ProvisionerNoOpProvision {
+	if f.clusterServiceClient.GetProvisionerNoOpProvision() {
 		additionalProperties["provisioner_noop_provision"] = "true"
 	}
-	if f.clusterServiceConfig.ProvisionerNoOpDeprovision {
+	if f.clusterServiceClient.GetProvisionerNoOpDeprovision() {
```

Comment:
This is totally optional for this pull request but I was thinking since `ClusterServiceClient` already has the optional cluster flags it would be cool to hide all this `additionalProperties` logic (including the hard-coded ones above) behind a `ClusterServiceClient` method that looks something like
```
func (csc *ClusterServiceClient) AddProperties(builder *cmv1.ClusterBuilder) *cmv1.ClusterBuilder
```
and then further down where we use `additionalProperties` just have
```
clusterBuilder = csc.AddProperties(clusterBuilder)
```

Reply:
Oh I like this. I'll do it in a separate PR for clarity

Reply:
Done here: https://github.com/Azure/ARO-HCP/pull/720